variants/esp32c3/variant.cpp

```diff
@@ -0,0 +1,13 @@
+#include "esp32-hal-rgb-led.h"
+
+extern void ARDUINO_ISR_ATTR __pinMode(uint8_t pin, uint8_t mode)
```

Comment:
why is this necessary? pinMode in the core should just return if pin is not valid.

Reply:
I was trying to implement this https://github.com/espressif/arduino-esp32/pull/6808#pullrequestreview-986096611

Reply:
I understand, I just question the necessity to have such code in the variant just to get something as common as neopixel on the board working.
@igrr any comments?

Something like this in the core's `__pinMode` would skip execution:
```cpp
if(pin > SOC_GPIO_PIN_COUNT && pin == LED_BUILTIN){
    return; // or pin -= SOC_GPIO_PIN_COUNT; to allow setting the mode.
}
```

Reply:
@PilnyTomas this is still outstanding. Requiring variant.cpp to blink a led is not desired situation.

Reply:
@igrr Could you please give us more info about why should we have board-specific code in the variants folder instead of having in the common file?
Should I keep it like this, or revert back to the common file?

Reply:
I was thinking that this is behavior we want only on specific boards which have a WS2812B, and where we'd like to provide such emulation.

Makers of other boards should be able implement some different logic, or opt out of such emulation. In the latter case, the code for emulating an LED ideally shouldn't be compiled or linked.

Reply:
if you do not have that LED defined, code will not be compiled anyway. requiring all boards with RGB leds to include a variant cpp for two lines is a bit too much IMO. There are plenty boards with RGB leds

Reply:
Although, given that you are introducing a BOARD_HAS_NEOPIXEL macro, maybe we can indeed move this functionality into the core.